lib/commons/dom/get-target-rects.js

```diff
@@ -0,0 +1,30 @@
+import findNearbyElms from './find-nearby-elms';
+import { splitRects, hasVisualOverlap } from '../math';
+import memoize from '../../core/utils/memoize';
+
+export default memoize(getTargetRects);
+
+/**
+ * Return all unobscured rects of a target.
+ * @see https://www.w3.org/TR/WCAG22/#dfn-bounding-boxes
+ * @param {VitualNode} vNode
+ * @return {DOMRect[]}
+ */
+function getTargetRects(vNode) {
+  const nodeRect = vNode.boundingClientRect;
+  const overlappingVNodes = findNearbyElms(vNode).filter(vNeighbor => {
+    return (
```

Comment:
You're going to need to filter the non-focusable descendants from this, just like we do in target-size-evaluate. Otherwise something like an icon link ends up with no size. Descendants are part of the target, they don't obscure it.

Reply:
Yep. We're proposing doing that work as part of https://github.com/dequelabs/axe-core/issues/4120 since target size evaluate has the code already and I doing the refactoring.

Reply:
We agreed that this should be brought into this pr by updating the `getTargetRects` function to filter out descendants that are not in tab order before calling `splitRects`.